qp_spades/qp_spades.py

```diff
@@ -0,0 +1,153 @@
+    -------
+    bool, list, str
+        The results of the job
+    """
+    # Step 1 get the rest of the information need to run Bowtie2
+    qclient.update_job_step(job_id, "Step 1 of 4: Collecting information")
+    artifact_id = parameters['input']
+    del parameters['input']
+
+    # Get the artifact filepath information
+    artifact_info = qclient.get("/qiita_db/artifacts/%s/" % artifact_id)
+    fwd_seqs = sorted(artifact_info['files']['raw_forward_seqs'])
+    if 'raw_reverse_seqs' in artifact_info['files']:
+        rev_seqs = sorted(artifact_info['files']['raw_reverse_seqs'])
+    else:
+        return False, None, 'This plugin expects forward and reverse reads.'
+
+    if len(fwd_seqs) != len(rev_seqs):
+        mgs = 'There is a different number of forward and reverse read files.'
+        return False, None, mgs
+
+    # Get the artifact metadata
+    prep_info = qclient.get('/qiita_db/prep_template/%s/'
+                            % artifact_info['prep_information'][0])
+    df = pd.read_csv(prep_info['prep-file'], sep='\t', dtype='str',
+                     na_values=[], keep_default_na=True)
+    if 'run_prefix' not in df.columns:
+        return False, None, 'Missing run_prefix column in your preparation'
+    prefix_to_name = df.set_index('run_prefix')['sample_name'].to_dict()
+
+    # Step 2 generating command
+    qclient.update_job_step(job_id, "Step 2 of 4: Generating commands")
+
+    flash_cmd = (f'flash --threads {parameters["threads"]} --max-overlap=%d '
+                 f'--max-mismatch-density=0.1 --output-directory {out_dir} '
+                 '--output-prefix="%s" %s %s')
+    spades_cmd = (
+        f'spades.py --{parameters["type"]} -t {parameters["threads"]} '
+        f'-m {parameters["memory"]} -k {parameters["k-mers"]} -1 %s -2 %s '
+        f'-o {out_dir}/%s')
+
+    commands = []
+    outfiles = []
+    overlap = None
+
+    for fwd_fp, rev_fp in zip(fwd_seqs, rev_seqs):
+        fwd_fn = basename(fwd_fp)
+
+        run_prefix = None
+        for rp in prefix_to_name:
+            if fwd_fn.startswith(rp):
+                if run_prefix is None:
+                    run_prefix = rp
+                else:
+                    msg = ('Multiple run prefixes match this fwd '
+                           'file: {fwd_fn}')
+                    return False, None, msg
+
+        rev_fn = basename(rev_fp)
+        # if we have reverse reads, make sure the matching pair also
+        # matches the run prefix:
+        if not rev_fn.startswith(run_prefix):
+            msg = ('Reverse read does not match this run prefix.\nRun prefix: '
+                   '%s\nForward read: %s\nReverse read: %s\n' % (
+                    run_prefix, fwd_fn, rev_fn))
+            return False, None, msg
+
+        sample_name = prefix_to_name[run_prefix]
+        out_fp = f'{out_dir}/{sample_name}.fasta'
+        if parameters['merging'] == 'no merge':
+            cmd = (f'{spades_cmd} > {out_dir}/%s.spades.log 2>&1; '
```

Comment:
I would suggest to make this `&&` so that the mv command is not run unless this command exits with 0 status. Otherwise fi the files are not there after the command runs and fails we'll see an unhelpful message saying something like `mv file doesn't exist`. What do you think?

```suggestion
            cmd = (f'{spades_cmd} > {out_dir}/%s.spades.log 2>&1&& '
```